### ISSUE TITLE
False positive for "unsupported right operand type" for enum.Enum _value2member_map_

### ISSUE THREAD

**Issue by jamesbraza** [python/mypy#9711]
**Bug Report**

Currently, mypy is warning about an "unsupported right operand type" when checking if a value is `in` an `enum.Enum`'s `_value2member_map_`.  I believe this is a false positive error.

**To Reproduce**

```python
"""Show that mypy might not grok enum's _value2member_map_."""

from enum import Enum


class Foo(Enum):
    """Enum class whose values we care about."""

    BAR = 0
    BAZ = 1

    @classmethod
    def has_value(cls, val: int) -> bool:
        """Get if a value is in this enum.

        SEE: https://stackoverflow.com/a/43634746/11163122

        """
        return val in cls._value2member_map_  # Location of the false positive
```

**Expected Behavior**

Since `enum.Enum` is built into the standard library, I would expect mypy not to raise an error about a check of `_value2member_map_`.

**My Unsolicited Thoughts**

From looking at the source, it seems `_value2member_map_` is set as an empty dictionary in `Enum`'s meta class `EnumMeta` within the `__new__` method.  This may explain why mypy is not properly understanding it.

_Disclaimer_: I am not sure how mypy handles runtime creations of the standard library within `__new__`.

**Actual Behavior**

Here's the output of mypy:

```none
has_value_enum.py:19:16: error: Unsupported right operand type for in ("Foo")  [operator]
```

Aside: the error message "operand type for in" doesn't entirely make sense to me.

**Your Environment**

- Mypy version used: `0.790`
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): nothing special
- Python version used: `3.8.6`
- Operating system and version: macOS Catalina, version 10.15.7

**Comment by JelleZijlstra** [comment]
We just need to add this to the typeshed stub for enum. Presumably otherwise mypy thinks `Foo._value2member_map_` is an instance of `Foo`.
